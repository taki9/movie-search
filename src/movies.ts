export default {
  count: 5042,
  next: 'http://imdb.hriks.com/movie/?limit=50&offset=50',
  previous: null,
  results: [
    {
      id: 2844,
      title: 'Justin Bieber: Never Say Never',
      date_of_release: '2011-01-01',
      rating: 1.6,
      tags: ['documentary', 'music'],
      director: '7745: Jon M. Chu',
    },
    {
      id: 1146,
      title: 'Foodfight! ',
      date_of_release: '2012-01-01',
      rating: 1.7,
      tags: ['action', 'fantasy', 'animation', 'comedy', 'family'],
      director: '8202: Lawrence Kasanoff',
    },
    {
      id: 2278,
      title: 'Disaster Movie ',
      date_of_release: '2008-01-01',
      rating: 1.9,
      tags: ['comedy'],
      director: '8418: Jason Friedberg',
    },
    {
      id: 2305,
      title: 'Superbabies: Baby Geniuses 2 ',
      date_of_release: '2004-01-01',
      rating: 1.9,
      tags: ['sci-fi', 'comedy', 'family'],
      director: '8701: Bob Clark',
    },
    {
      id: 4615,
      title: 'The Helix... Loaded ',
      date_of_release: '2005-01-01',
      rating: 1.9,
      tags: ['action', 'adventure', 'fantasy', 'sci-fi', 'comedy'],
      director: '9978: A. Raven Cruz',
    },
    {
      id: 1739,
      title: 'United Passions ',
      date_of_release: '2014-01-01',
      rating: 2,
      tags: ['drama', 'history', 'sport'],
      director: '8463: Frédéric Auburtin',
    },
    {
      id: 3515,
      title: "Who's Your Caddy? ",
      date_of_release: '2007-01-01',
      rating: 2,
      tags: ['comedy', 'sport'],
      director: '9341: Don Michael Paul',
    },
    {
      id: 2993,
      title: 'From Justin to Kelly ',
      date_of_release: '2003-01-01',
      rating: 2.1,
      tags: ['romance', 'comedy', 'musical'],
      director: '8838: Robert Iscove',
    },
    {
      id: 3350,
      title: 'Glitter ',
      date_of_release: '2001-01-01',
      rating: 2.1,
      tags: ['romance', 'drama', 'music'],
      director: '3775: Vondie Curtis-Hall',
    },
    {
      id: 3674,
      title: 'Crossover ',
      date_of_release: '2006-01-01',
      rating: 2.1,
      tags: ['action', 'sport'],
      director: '8991: Preston A. Whitmore II',
    },
    {
      id: 329,
      title: 'Son of the Mask ',
      date_of_release: '2005-01-01',
      rating: 2.2,
      tags: ['fantasy', 'comedy', 'family'],
      director: '7826: Lawrence Guterman',
    },
    {
      id: 4629,
      title: 'Snow White: A Deadly Summer ',
      date_of_release: '2012-01-01',
      rating: 2.2,
      tags: ['horror'],
      director: '9988: David DeCoteau',
    },
    {
      id: 4778,
      title: 'Subconscious ',
      date_of_release: '2015-01-01',
      rating: 2.2,
      tags: ['action', 'thriller', 'mystery'],
      director: '10086: Georgia Hilton',
    },
    {
      id: 2202,
      title: 'Epic Movie ',
      date_of_release: '2007-01-01',
      rating: 2.3,
      tags: ['adventure', 'comedy'],
      director: '8418: Jason Friedberg',
    },
    {
      id: 2323,
      title: 'Alone in the Dark ',
      date_of_release: '2005-01-01',
      rating: 2.3,
      tags: ['sci-fi', 'horror'],
      director: '8710: Uwe Boll',
    },
    {
      id: 3535,
      title: 'Chairman of the Board ',
      date_of_release: '1998-01-01',
      rating: 2.3,
      tags: ['comedy'],
      director: '9353: Alex Zamm',
    },
    {
      id: 505,
      title: 'Battlefield Earth ',
      date_of_release: '2000-01-01',
      rating: 2.4,
      tags: ['action', 'adventure', 'sci-fi'],
      director: '7904: Roger Christian',
    },
    {
      id: 909,
      title: 'Gigli ',
      date_of_release: '2003-01-01',
      rating: 2.4,
      tags: ['romance', 'comedy', 'crime'],
      director: '7870: Martin Brest',
    },
    {
      id: 2862,
      title: 'Baby Geniuses ',
      date_of_release: '1999-01-01',
      rating: 2.5,
      tags: ['sci-fi', 'comedy', 'family', 'crime'],
      director: '8992: Bob Clark',
    },
    {
      id: 4884,
      title: 'Sunday School Musical ',
      date_of_release: '2008-01-01',
      rating: 2.5,
      tags: ['musical', 'drama'],
      director: '10149: Rachel Goldenberg',
    },
    {
      id: 4143,
      title: '30 Nights of Paranormal Activity with the Devil Inside the Girl with the Dragon Tattoo ',
      date_of_release: '2013-01-01',
      rating: 2.6,
      tags: ['comedy'],
      director: '9705: Craig Moss',
    },
    {
      id: 4535,
      title: 'The Vatican Exorcisms ',
      date_of_release: '2013-01-01',
      rating: 2.6,
      tags: ['horror'],
      director: '6666: Joe Marino',
    },
    {
      id: 1082,
      title: 'Inchon ',
      date_of_release: '1981-01-01',
      rating: 2.7,
      tags: ['drama', 'history', 'war'],
      director: '8175: Terence Young',
    },
    {
      id: 1664,
      title: 'Meet the Spartans ',
      date_of_release: '2008-01-01',
      rating: 2.7,
      tags: ['comedy'],
      director: '8418: Jason Friedberg',
    },
    {
      id: 1712,
      title: 'Dragonball: Evolution ',
      date_of_release: '2009-01-01',
      rating: 2.7,
      tags: ['action', 'adventure', 'fantasy', 'sci-fi', 'thriller'],
      director: '8161: James Wong',
    },
    {
      id: 2192,
      title: 'Date Movie ',
      date_of_release: '2006-01-01',
      rating: 2.7,
      tags: ['romance', 'comedy'],
      director: '8650: Aaron Seltzer',
    },
    {
      id: 3933,
      title: 'The Real Cancun ',
      date_of_release: '2003-01-01',
      rating: 2.7,
      tags: ['documentary'],
      director: '9578: Rick de Oliveira',
    },
    {
      id: 4482,
      title: 'The 41-Year-Old Virgin Who Knocked Up Sarah Marshall and Felt Superbad About It ',
      date_of_release: '2010-01-01',
      rating: 2.7,
      tags: ['comedy'],
      director: '9705: Craig Moss',
    },
    {
      id: 2076,
      title: 'Jaws: The Revenge ',
      date_of_release: '1987-01-01',
      rating: 2.8,
      tags: ['adventure', 'thriller', 'horror'],
      director: '8607: Joseph Sargent',
    },
    {
      id: 2311,
      title: 'Doogal ',
      date_of_release: '2006-01-01',
      rating: 2.8,
      tags: ['adventure', 'fantasy', 'animation', 'comedy', 'family'],
      director: '8703: Dave Borthwick',
    },
    {
      id: 2338,
      title: 'Marci X ',
      date_of_release: '2003-01-01',
      rating: 2.8,
      tags: ['comedy', 'music'],
      director: '8576: Richard Benjamin',
    },
    {
      id: 2591,
      title: 'Steel ',
      date_of_release: '1997-01-01',
      rating: 2.8,
      tags: ['action', 'sci-fi', 'crime'],
      director: '8848: Kenneth Johnson',
    },
    {
      id: 2706,
      title: "Barney's Great Adventure ",
      date_of_release: '1998-01-01',
      rating: 2.8,
      tags: ['adventure', 'family'],
      director: '8907: Steve Gomer',
    },
    {
      id: 3028,
      title: "Dancin' It's On ",
      date_of_release: '2015-01-01',
      rating: 2.8,
      tags: ['family', 'musical'],
      director: '9079: David Winters',
    },
    {
      id: 4309,
      title: 'Hum To Mohabbat Karega ',
      date_of_release: '2000-01-01',
      rating: 2.8,
      tags: ['action', 'thriller', 'romance', 'comedy'],
      director: '9798: Kundan Shah',
    },
    {
      id: 4453,
      title: 'Mutant World ',
      date_of_release: '2014-01-01',
      rating: 2.8,
      tags: ['sci-fi'],
      director: '9885: David Winning',
    },
    {
      id: 4623,
      title: 'Frat Party ',
      date_of_release: '2009-01-01',
      rating: 2.8,
      tags: ['comedy'],
      director: '9985: Robert Bennett',
    },
    {
      id: 2008,
      title: 'BloodRayne ',
      date_of_release: '2005-01-01',
      rating: 2.9,
      tags: ['action', 'adventure', 'fantasy', 'horror'],
      director: '8585: Uwe Boll',
    },
    {
      id: 2019,
      title: "The True Story of Puss'N Boots ",
      date_of_release: '2009-01-01',
      rating: 2.9,
      tags: ['adventure', 'animation', 'comedy'],
      director: '3127: Jérôme Deschamps',
    },
    {
      id: 3610,
      title: 'Daddy Day Camp ',
      date_of_release: '2007-01-01',
      rating: 2.9,
      tags: ['comedy', 'family'],
      director: '9398: Fred Savage',
    },
    {
      id: 630,
      title: 'Rollerball ',
      date_of_release: '2002-01-01',
      rating: 3,
      tags: ['action', 'sci-fi', 'sport'],
      director: '7771: John McTiernan',
    },
    {
      id: 2532,
      title: 'Shanghai Surprise ',
      date_of_release: '1986-01-01',
      rating: 3,
      tags: ['adventure', 'romance', 'drama', 'crime'],
      director: '8816: Jim Goddard',
    },
    {
      id: 4091,
      title: 'Phat Girlz ',
      date_of_release: '2006-01-01',
      rating: 3,
      tags: ['comedy'],
      director: '9673: Nnegest Likké',
    },
    {
      id: 4534,
      title: 'The Dead Undead ',
      date_of_release: '2010-01-01',
      rating: 3,
      tags: ['action', 'sci-fi', 'horror'],
      director: '6663: Matthew R. Anderson',
    },
    {
      id: 5030,
      title: 'The Ridges ',
      date_of_release: '2011-01-01',
      rating: 3,
      tags: ['thriller', 'drama', 'horror'],
      director: '7586: Brandon Landers',
    },
    {
      id: 2409,
      title: 'Left Behind ',
      date_of_release: '2014-01-01',
      rating: 3.1,
      tags: ['action', 'fantasy', 'thriller', 'mystery', 'drama'],
      director: '8754: Vic Armstrong',
    },
    {
      id: 2521,
      title: 'Larry the Cable Guy: Health Inspector ',
      date_of_release: '2006-01-01',
      rating: 3.1,
      tags: ['romance', 'comedy'],
      director: '8808: Trent Cooper',
    },
    {
      id: 3455,
      title: 'Witless Protection ',
      date_of_release: '2008-01-01',
      rating: 3.1,
      tags: ['comedy', 'crime'],
      director: '9317: Charles Robert Carner',
    },
    {
      id: 4032,
      title: 'Lords of London ',
      date_of_release: '2014-01-01',
      rating: 3.1,
      tags: ['thriller', 'mystery', 'crime'],
      director: '9641: Antonio Simoncini',
    },
  ],
};
