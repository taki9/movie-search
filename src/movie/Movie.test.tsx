import { expect, describe, it, afterEach } from 'vitest';
import { cleanup, render } from '@testing-library/react';
import { MovieCard } from '.';
import movies from '../movies';

const movie = movies.results[2];
const renderComponent = () => render(<MovieCard movie={movie} />);

describe('MovieCard component', () => {
  afterEach(cleanup);

  it('should match the snapshot', () => {
    const { container } = renderComponent();

    expect(container.innerHTML).toMatchSnapshot();
  });

  it('should render the movie title', () => {
    const { getByText } = renderComponent();

    const title = getByText(movie.title.trim());
    expect(title).toBeTruthy();
  });

  it('should render the release year', () => {
    const { getByText } = renderComponent();
    const expectedYear = new Date(movie.date_of_release).getFullYear();

    const year = getByText(expectedYear.toString());

    expect(year).toBeTruthy();
  });
});
