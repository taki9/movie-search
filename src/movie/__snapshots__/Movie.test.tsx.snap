// Vitest Snapshot v1

exports[`MovieCard component > should match the snapshot 1`] = `"<div class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-oyb9jb-MuiPaper-root-MuiCard-root\\"><div class=\\"MuiBox-root css-vtpeil\\"><img class=\\"MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root\\" src=\\"https://via.placeholder.com/300x450\\" height=\\"450\\" alt=\\"green iguana\\"><div class=\\"MuiCardContent-root css-16yqspb-MuiCardContent-root\\"><h2 class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-ua6pm7-MuiTypography-root\\">Disaster Movie </h2><div class=\\"css-1js6t29-MuiStack-root\\"><p class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\">2008</p><div class=\\"MuiBox-root css-1ibfoaj\\"></div><p class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\">PG</p><div class=\\"MuiBox-root css-1ibfoaj\\"></div><p class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\">2h 7m</p></div><p class=\\"MuiTypography-root MuiTypography-body1 css-1ff1qco-MuiTypography-root\\">In Steven Spielberg's massive blockbuster, paleontologists Alan Grant (Sam Neill) and Ellie Sattler (Laura Dern) and mathematician Ian Malcolm (Jeff Goldblum) are among a select group chosen to tour an island theme park populated by dinosaurs created from prehistoric DNA. While the park's mastermind, billionaire John Hammond (Richard Attenborough), assures everyone that the facility is safe, they find out otherwise when various ferocious predators break free and go on the hunt.</p><div class=\\"css-14vtlsk-MuiStack-root\\"><div class=\\"MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-9hjqux-MuiChip-root\\"><span class=\\"MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label\\">comedy</span></div></div></div></div></div>"`;
